lib/plugins/aws/provider/awsProvider.js

```diff
@@ -19,7 +19,7 @@ const constants = {
   providerName: 'aws',
 };
 
-const validAPIGatewayStageNamePattern = /^[a-zA-Z0-9_]+$/;
+const validAPIGatewayStageNamePattern = /^[-:a-zA-Z0-9_{}$.',\s]+$/;
```

Comment:
Won't that make be able to do something like: `stage: ${opt:stage, 'app:staging'}` or `stage: ${opt:stage, 'app,staing'}`?

Are those allowed? I think we should att some more weird cases like that in the test